ci/jenkins/README.md

```diff
@@ -1 +1,97 @@
 # Nephe CI: Jenkins
+
+## Table of Contents
+
+<!-- toc -->
+- [Introduction](#introduction)
+- [Jenkins Deployment](#jenkins-deployment)
+  - [Teraform](#teraform)
+  - [Ansible](#ansible)
+  - [Jenkins Job](#jenkins-job)
+    - [Requirements](#requirements)
+    - [Apply the jobs](#apply-the-jobs)
+    - [Jenkins job updater](#jenkins-job-updater)
+- [List of jenkins job](#list-of-jenkins-job)
+<!-- /toc -->
+
+## Introduction
+
+This directory includes all the scripts required to run CI for nephe.
+
+We have a static VM configured to run as jenkins node.  When a jenkins job is
+triggered, the nephe code will be cloned on the static VM. The jenkins node will
+call terraform to a deploy dynamic VM and use ansible for installing any missing
+packages like docker. Then tests will be run on the dynamic VM, the VM will be
+destroyed once the test succeeds or fails.
+
+## Jenkins Deployment
+
+### Teraform
+
+We create terrform tfvars file in nephe-ci.sh with jenkins credetials. [deploy.sh](./deploy.sh)
+is the script to create vm, and store the terraform state in a specified directory,
+which can be used to get VM information later. [destroy.sh](./destroy.sh) helps to detroy the VM
+and delete terraform workspace as well.
+
+For more details about terraform, please refer [here](https://www.terraform.io/docs).
+
+### Ansible
+
+nsible will help us for system configuration after the VM installation finishes, all
+ansible tasks are in playbook/roles. For example, we add task to install docker in
+[playbook/roles/docker/tasks/main.yml](playbook/roles/docker/tasks/main.yml).
+
+Ansile also help to copy ssh keys to dynamic VM, by which we can run tests and collect
+log later. The ssh-keys are generated on static VM if nephe-ci checks there is no
+ssh-keys.
+
+For more details about ansible, please refer [here](https://docs.ansible.com/).
+
+### Jenkins Job
+
+We use jenkins job-builder to manage jenkins jobs, the codes are in the directory jobs.
+
+#### Requirements
+
+Yaml files under current directory can be generated via
+jenkins-job-builder. If you want to try out the tests on your local jenkins
+setup, please notice the following requirements:
+
+* Jenkins setup
```

Comment:
Where do you point to local Jenkins IP/User/Token? Isnt it done via some export.